src/background.js

```diff
@@ -1,18 +1,66 @@
-chrome.tabs.onUpdated.addListener(function(tabId, changeInfo) {
-  if (changeInfo.status !== 'complete') { return; }
+if (PLATFORM == 'chrome' || PLATFORM == 'firefox') {
+  chrome.tabs.onUpdated.addListener(function(tabId, changeInfo) {
+    if (changeInfo.status !== 'complete') { return; }
 
-  chrome.tabs.executeScript(tabId, {code: 'window.uiPreviewerButtonInjected;'}, function(res) {
-    if (res && !res[0]) {
-      chrome.tabs.executeScript(tabId, {code: 'window.uiPreviewerButtonInjected = true;'});
+    if (PLATFORM == 'chrome') {
+      chrome.tabs.executeScript(tabId, {code: 'window.uiPreviewerButtonInjected;'}, function(res) {
+        if (res && !res[0]) {
+          // chrome.tabs.executeScript(tabId, {code: 'window.uiPreviewerButtonInjected = true;'});
+          chrome.tabs.executeScript(tabId, {file: 'src/ui-previewer-button.js'});
+        }
+      });
+    } else if(PLATFORM == 'firefox') {
```

Comment:
The platform is only Chrome or Firefox in this context, might be worth keeping this to be explicit anyway though.